alpha/engagements/2023/rust/update-2023-11.md

```diff
@@ -0,0 +1,57 @@
+This month, the Rust Infrastructure threat model became [available]() and there has been some security-based action in the crates.io ecosystem.
+
+## Threat Modeling
+
+Here is the progress on the four threat models being developed by the Rust Foundation:
+
+1. **Crates ecosystem**: [Published](https://drive.google.com/file/d/1YxpJ0W5eqat2Y3ZfbdwKm_AoNhX3hIj_/).
+2. **Rust Infrastructure**: [Published]()
+3. **crates.io**: Potential threats identified. Filling out the details.
+4. **Rust Project**: Threat outline developed.
+
+## Engineering
+
+The technical staff at the Rust Foundation continues to work on engineering efforts directly related to security:
+
+### X-Forwarded-For Header Processing
+
+Tobias [implemented](https://github.com/rust-lang/crates.io/pull/7359) [`ngx_http_realip_module`](http://nginx.org/en/docs/http/ngx_http_realip_module.html) directly into the axum server, further reducing the need for custom `nginx` wrappers. Security-wise, Tobias also found that Heroku wasn't handling the `x-forwarded-for` headers correctly. Instead of adding the connecting IP to the last header, or merging the headers, it adds the IP to the first header. This was reported to Heroku and the hope is that they fix it, but, for now, the PR implements a workaround.
+
+### crates.io Request Failure Incident
+
+crates.io [experienced](https://status.crates.io/incidents/fsrmr4bk8trw) some failing requests and timeouts on October 18 due to an unexpected increase in traffic. While this looks to have been legitimate downloads requests spiking, Tobias is leading the investigation to ensure there is no major security issue.
+
+### Replace Ring with p256
+
+Tobias has [moved](https://github.com/rust-lang/crates.io/pull/7252) the crates.io dependency on `ring` over to `p256`. The `ring` project is releasing new versions without tags on the corresponding repository or even changelog entries, which is leaving us in the dark of what has changed or what is new. `p256` is a pure-Rust alternative for the only ring algorithm that we're currently using, and it also includes functions to parse the PEM format public keys.
+
+### Reading Dependencies from Cargo.toml
+
+To avoid the [hole](https://blog.vlt.sh/blog/the-massive-hole-in-the-npm-ecosystem) found in the `npm` ecosystem, Tobias is [proactively ensuring](https://github.com/rust-lang/crates.io/pull/7238) that all dependencies are read from an embedded cargo.toml file as opposed to a JSON metadata blog.
```

Comment:
```suggestion
To avoid the [hole](https://blog.vlt.sh/blog/the-massive-hole-in-the-npm-ecosystem) found in the `npm` ecosystem, Tobias is [proactively ensuring](https://github.com/rust-lang/crates.io/pull/7238) that all dependencies are read from an embedded `Cargo.toml` file as opposed to a JSON metadata blog.
```